pkg/globalnet/controllers/global_ingressip_controller.go

```diff
@@ -165,29 +162,47 @@ func (c *globalIngressIPController) onCreate(ingressIP *submarinerv1.GlobalIngre
 	}
 
 	if ingressIP.Spec.Target == submarinerv1.ClusterIPService {
-		chainName := ingressIP.GetAnnotations()[kubeProxyIPTableChainAnnotation]
-		if chainName == "" {
+		serviceRef := ingressIP.Spec.ServiceRef
+
+		service, exists, err := getService(serviceRef.Name, ingressIP.Namespace, c.services, c.scheme)
+		if err != nil || !exists {
 			_ = c.pool.Release(ips...)
+			key := fmt.Sprintf("%s/%s", ingressIP.Namespace, serviceRef.Name)
+			klog.Warningf("Exported Service %q does not exist yet - re-queueing", key)
 
-			klog.Warningf("%q annotation is missing on %q", kubeProxyIPTableChainAnnotation, key)
+			return false
+		}
 
-			return true
+		internalService := &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      GetInternalSvcName(service.Name),
+				Namespace: service.Namespace,
+				Labels: map[string]string{
+					ServiceRefLabel: service.Name,
+				},
+			},
 		}
 
-		err = c.iptIface.AddIngressRulesForService(ips[0], chainName)
-		if err != nil {
-			klog.Errorf("Error while programming Service %q ingress rules: %v", key, err)
+		extIPs := []string{ips[0]}
+		internalService.Spec.Ports = service.Spec.Ports
+		internalService.Spec.Selector = service.Spec.Selector
+		internalService.Spec.ExternalIPs = extIPs
 
+		_, err = createService(internalService, c.services)
+		if err != nil {
 			_ = c.pool.Release(ips...)
+			key := fmt.Sprintf("%s/%s", internalService.Namespace, internalService.Name)
+			klog.Warningf("Failed to create the internal Service %q ", key)
 
+			_ = c.pool.Release(ips...)
 			ingressIP.Status.Conditions = util.TryAppendCondition(ingressIP.Status.Conditions, &metav1.Condition{
 				Type:    string(submarinerv1.GlobalEgressIPAllocated),
 				Status:  metav1.ConditionFalse,
-				Reason:  "ProgramIPTableRulesFailed",
-				Message: fmt.Sprintf("Error programming ingress rules: %v", err),
+				Reason:  "FailedToCreateInternalGlobalnetService",
```

Comment:
We can shorten this a bit.

```suggestion
				Reason:  "InternalServiceCreationFailed",
```